hazelcast/src/main/java/com/hazelcast/map/impl/BasicMapContextQuerySupport.java

```diff
@@ -142,9 +142,10 @@ public Set queryLocalMember(String mapName, Predicate predicate, IterationType i
             if (partitionIds.isEmpty()) {
                 return result;
             }
-        } catch (QueryResultSizeExceededException e) {
-            throw ExceptionUtil.rethrow(e);
         } catch (Throwable t) {
+            if (t.getCause() instanceof QueryResultSizeExceededException) {
```

Comment:
How do you feel about modifying addResultsOfPagingPredicate and addResultsOfPredicate so that they throw only Runtime exceptions like queryOnPartitions? This way we can leave the catching part as is, which is , I think , more readable. (Also same for the collectResults method in TransactionalMapProxySupport)